python/cudf/cudf/core/column/column.py

```diff
@@ -599,17 +599,12 @@ def __setitem__(self, key: Any, value: Any):
             )
         else:
             try:
-                if is_scalar(value):
-                    input = self
-                    out = input.as_frame()._scatter(key, [value])._as_column()
-                else:
-                    if not isinstance(value, Column):
-                        value = as_column(value)
-                    out = (
-                        self.as_frame()
-                        ._scatter(key, value.as_frame())
-                        ._as_column()
-                    )
+                if not isinstance(key, Column):
+                    key = as_column(key)
+                if not is_scalar(value) and not isinstance(value, Column):
+                    value = as_column(value)
+                out = libcudf.copying.scatter(value, key, self)
+                out._with_type_metadata(self)
```

Comment:
This is an out-of-place function I think

Reply:
As in you don't think this call is necessary?